elements/lisk-codec/src/string.ts

```diff
@@ -14,12 +15,17 @@
 import { readBytes, writeBytes } from './bytes';
 
 export const writeString = (value: string): Buffer => {
-	const stringBuffer = Buffer.from(value, 'utf8');
+	const normalized = value.normalize('NFC');
+	const stringBuffer = Buffer.from(normalized, 'utf8');
 
 	return writeBytes(stringBuffer);
 };
 
 export const readString = (buffer: Buffer, offset: number): [string, number] => {
 	const [value, size] = readBytes(buffer, offset);
-	return [value.toString('utf8'), size];
+	const decodedStr = new TextDecoder('utf8', { fatal: true }).decode(value);
```

Comment:
What is the TextDecoder doing here? Why can't we say `const decodedStr = value.toString('utf8')` since we check for equality with the normalized string in the next line anyway?

Reply:
the TextDecoder is also checking validity of the bytes using `fatal: true`. For example, `0x80` will throw error